services/py-api/src/utils.py

```diff
@@ -43,3 +45,40 @@ async def stop(self) -> None:
             raise RuntimeError("The AsyncClient has not been initialized!")
 
         await self._async_client.aclose()
+
+
+class JwtUtility:
+    """A class providing methods for encoding data in with a predefined format into a JWT token, or decoding a JWT
+    token into a predefined format (TypedDict)"""
+
+    @staticmethod
+    def encode_data(data: JwtUserData) -> str:
+        return str(encode(data, key="secret", algorithm="HS256"))
+
+    @staticmethod
+    def decode_data(token: str) -> Result[JwtUserData, str]:
+        try:
+            decoded_token: dict[str, Any] = decode(token, "secret", algorithms=["HS256"])
+
+            if "sub" in decoded_token and "is_admin" in decoded_token and "team_id" in decoded_token:
+                return Ok(cast(JwtUserData, decoded_token))
+
+            LOG.warning(
```

Comment:
I would avoid logging all of these. They do not really provide any value in deployed environments as it's expected that these are user errors.

In the case they're not user facing errors, use `LOG.debug`

Reply:
Are they not user-facing errors? For example, showing that the JWT has expired is a thing you want to show your user. What is your definition of user-facing errors?

Reply:
Reread my comment